tests/sentry/api/endpoints/test_release_threshold.py

```diff
@@ -147,3 +148,87 @@ def test_valid(self):
         assert data["environment"]["id"] == str(self.canary_environment.id)
         assert data["environment"]["name"] == self.canary_environment.name
         assert data["date_added"] is not None
+
+    def test_get_invalid_project(self):
+        url_with_invalid_project = reverse(
+            "sentry-api-0-project-release-thresholds",
+            kwargs={
+                "organization_slug": self.organization.slug,
+                "project_slug": "Why did it have to be snakes?",
+            },
+        )
+        response = self.client.get(url_with_invalid_project, data={"environment": "canary"})
+        assert response.status_code == 404
+
+    def test_get_invalid_environment(self):
+        response = self.client.get(self.url, data={"environment": "The Hovitos are near"})
+        assert response.status_code == 200
+        assert len(response.data) == 0
+
+    def test_get_valid_no_environment(self):
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        assert len(response.data) == 0
+        ReleaseThreshold.objects.create(
+            threshold_type=0,
+            trigger_type=2,
+            value=100,
+            window_in_seconds=1800,
+            project=self.project,
+            environment=self.canary_environment,
+        )
+
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        assert len(response.data) == 1
+
+        created_threshold = response.data[0]
+
+        assert created_threshold["threshold_type"] == "total_error_count"
+        assert created_threshold["trigger_type"] == "absolute_over"
+        assert created_threshold["value"] == 100
+        assert created_threshold["window_in_seconds"] == 1800
+        assert created_threshold["project"]["id"] == str(self.project.id)
+        assert created_threshold["project"]["slug"] == self.project.slug
+        assert created_threshold["project"]["name"] == self.project.name
+        assert created_threshold["environment"]["id"] == str(self.canary_environment.id)
+        assert created_threshold["environment"]["name"] == self.canary_environment.name
```

Comment:
nit: you could also check for dictionary equality

Reply:
Not 100% sure our dictionary for `environment` and `project` are equal to what comes back from the endpoint (since they go through the outbound serializers)